rapid7_insightvm/komand_rapid7_insightvm/util/resource_helpers.py

```diff
@@ -0,0 +1,182 @@
+        :param logger: Logger object available to Komand actions/triggers, usually self.logger
+        :return: ResourceHelper object
+        """
+        self.logger = logger
+        self.session = session
+
+        # Suppress insecure request messages
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+    def v1_authenticate(self, console_url: str):
+        """
+        Authenticates to APIv1 and saves the session with appropriate headers for making v1 requests
+        :param console_url: URL to the InsightVM console
+        """
+        headers = {'Content-Type': 'text/xml'}
+        login_request = f'<?xml version="1.0" encoding="UTF-8"?><LoginRequest user-id="{self.session.auth.username}" password="{self.session.auth.password}"/>'
+        response = self.session.post(f"{console_url}/api/1.1/xml", data=login_request, verify=False,
+                                     headers=headers, allow_redirects=False)
+        if response.status_code == 200:
+            xml_response = etree.fromstring(response.text)
+            session_id = xml_response.attrib['session-id']
+            headers = {
+                "nexposeCCSessionID": session_id,
+                "Cookie": f"nexposeCCSessionID={session_id}"
+            }
+            self.logger.info('Successfully authenticated to APIv1')
+            self.session.headers.update(headers)
+        elif response.status_code == 302:
+            error = "ERROR: Authentication for APIv1 was not valid"
+            raise Exception(error)
+        else:
+            error = "ERROR: Failed to obtain an APIv1 session ID."
+            raise Exception(error)
+
+    def v1_deauthenticate(self, console_url: str):
+        """
+        Deauthenticates an existing APIv1 session using session header data
+        :param console_url: URL to the InsightVM console
+        """
+        self.session.headers.update({'Content-Type': 'text/xml'})
+        session_id = self.session.headers.get('nexposeCCSessionID', '')
+        logout_request = f'<?xml version="1.0" encoding="UTF-8"?><LogoutRequest session-id="{session_id}"/>'
+        response = self.session.post(f"{console_url}/api/1.1/xml", data=logout_request, verify=False)
+
+        if response.status_code == 200:
+            xml_response = etree.fromstring(response.text)
+            status = xml_response.attrib['success']
+            if int(status) == 1:
+                self.logger.info('Successfully logged out of APIv1 session')
+            else:
+                self.logger.info('Session ID for logout not valid, removing APIv1 session ID headers')
+            self.session.headers.pop('nexposeCCSessionID')
+            self.session.headers.pop('Cookie')
+        else:
+            error = "ERROR: Failed to log out via APIv1"
+            raise Exception(error)
+
+
+class ValidateUser:
+    """
+    Validates that a user exists and has the correct permissions
+    """
+    def __init__(self, session, logger):
```

Comment:
```suggestion
    def __init__(self, session: requests.Session, logger: logging.Logger):
```
Type hinting would be pretty helpful for users here